assets/javascripts/kitten/components/grid/side-layout.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import classNames from 'classnames'
+import bowser from 'bowser'
+import { SideGrid,
+         SideGridContent,
+         SideGridAside } from 'kitten/components/grid/side-grid'
+import { Grid, GridCol } from 'kitten/components/grid/grid'
+import DocLinkBox from 'kitten/components/box/doc-link-box'
+import { Title } from 'kitten/components/typography/title'
+
+export class SideLayout extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { needsFullHeight: false }
+  }
+
+  needsFullHeight() {
+    return !!bowser.safari
```

Comment:
C'est pour ça qu'on utilise `componentDidMount`. L'état `needsFullHeight` est à `false` par défaut, du coup le server et le front rendront le même composant. Une fois monté c'est là qu'on appelle `this. needsFullHeight` pour savoir si on est sur safari ou non afin de mettre à jour le `state` `needsFullHeight` :).